sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -406,6 +406,53 @@ ssize_t coap_opt_get_string(coap_pkt_t *pkt, uint16_t optnum,
     return (int)(max_len - left);
 }
 
+int coap_iterate_uri_query(coap_pkt_t *pkt, void **opt_pos,
+                           char *key, size_t key_len_max,
+                           char *value, size_t value_len_max)
+{
+    int len;
+    void *key_data = coap_iterate_option(pkt, COAP_OPT_URI_QUERY,
+                                         (uint8_t **)opt_pos, &len);
+    if (!key_data) {
+        return 0;
+    }
+
+    const char *value_data = memchr(key_data, '=', len);
+
+    size_t key_len, value_len;
+
+    if (value_data) {
+        key_len = (uintptr_t)value_data - (uintptr_t)key_data;
+        value_data += 1;
+        value_len = len - key_len - 1;
+    } else {
+        key_len = len;
+        value_data = NULL;
+        value_len = 0;
+
+        if (value && value_len_max > 1) {
```

Comment:
Thats' a weird thing to set; I've never seen a CoAP API where the absence of a value is treated as 1. (That's the convention with C `#define`s, but I doubt it's worth emulating).

Reply:
True, I could also just put `value[0] = 0`, just thought this might be a bit more convenient.